bundles/org.openhab.binding.dali/src/main/java/org/openhab/binding/dali/internal/handler/DaliDeviceHandler.java

```diff
@@ -0,0 +1,155 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DaliDeviceHandler} handles commands for things of type Device and Group.
+ *
+ * @author Robert Schmid - Initial contribution
+ */
+@NonNullByDefault
+public class DaliDeviceHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(DaliDeviceHandler.class);
+
+    static final int Switch100Percent = 254;
+
+    public DaliDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        initDeviceState();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE
+                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {
+            initDeviceState();
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    public void initDeviceState() {
+        Bridge bridge = getBridge();
+
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+        } else if (bridge.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (CHANNEL_DIM_AT_FADE_RATE.equals(channelUID.getId())
+                    || CHANNEL_DIM_IMMEDIATELY.equals(channelUID.getId())) {
+                Integer targetId = ((BigDecimal) this.thing.getConfiguration().get(TARGET_ID)).intValueExact();
+
+                DaliAddress address;
+                if (THING_TYPE_DEVICE.equals(this.thing.getThingTypeUID())) {
+                    address = DaliAddress.Short(targetId);
+                } else if (THING_TYPE_GROUP.equals(this.thing.getThingTypeUID())) {
+                    address = DaliAddress.Group(targetId);
+                } else {
+                    throw new DaliException("unknown device type");
+                }
+
+                boolean queryDeviceState = false;
+
+                if (command instanceof PercentType) {
+                    byte dimmValue = (byte) ((((PercentType) command).floatValue() * Switch100Percent) / 100);
+                    getBridgeHandler().sendCommand(new DaliDAPCCommand(address, dimmValue));
```

Comment:
What happens when the PercentType '0' command is sent, does it get handled correctly?